src/Tuttifrutti/Cache/Storage.hs

```diff
@@ -0,0 +1,47 @@
+module Tuttifrutti.Cache.Storage where
+
+import           Tuttifrutti.Prelude
+
+import           Data.Range.Range    (Range (..))
+
+-- | Storage implementation. This defines primitive operations, from which
+--   all others are derived.
+data Handle k p v m = Handle
```

Comment:
Let's spell out `k`, `p` and `v` at least here

Reply:
I ended up not changing the parameter names, but described them in the docstring. This way all the code base would be using the same letters and their meaning is documented here.